text/0513-app-specific-staging.md

```diff
@@ -0,0 +1,281 @@
+
+To use the new synthesizer:
+
+```ts
+import { AppStagingSynthesizer } from '@aws-cdk/app-staging-synthesizer';
+
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
+    appId: 'my-app-id', // put a unique id here
+    deploymentIdentities: DeploymentIdentities.defaultBootstrapRoles({ bootstrapRegion: 'us-east-1' }),
+
+    // How long to keep File and Docker assets around for rollbacks (without requiring resynth)
+    deployTimeFileAssetLifetime: Duration.days(100),
+    imageAssetVersionCount: 10,
+  }),
+});
+```
+
+For any additional customization (such as using custom buckets or ECR repositories), `DefaultStagingStack`
+can be subclasses or a full reimplementation of `IStagingResources` can be provided:
+
+```ts
+class MyStagingStack extends DefaultStaginStack {
+  private bucket?: s3.Bucket;
+
+  public addFile(asset: FileAssetSource): FileStagingLocation {
+    this.getCreateBucket();
+
+    return {
+      bucketName: 'my-asset-bucket',,
+      dependencyStack: this,
+    };
+  }
+
+  private createOrGetBucket() {
+    if (!this.bucket) {
+      this.bucket = new s3.Bucket(this, 'Bucket', {
+        bucketName: 'my-asset-bucket',
+      });
+    }
+    return this.bucket;
+  }
+}
+
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.customFactory({
+    factory: {
+      obtainStagingResources(stack, context) {
+        const myApp = App.of(stack);
+        return new MyStagingStack(myApp, `CustomStagingStack-${context.environmentString}`, {});
+      },
+    },
+  }),
+});
+```
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are launching a new synthesizer that has fewer demands on the AWS account that CDK apps are deployed into. It only
+needs preprovisioned Roles, and those are only necessary for CI/CD deployments or for cross-account deployments. For
+same-account, CLI deployments no bootstrapping is necessary anymore. If you are using bootstrapped roles anyway,
+they only need to be provisioned in one region, making it easier to use with StackSets.
+
+The new staging resources are specific to an application and can be cleaned up alongside the application. In addition,
+the way the staging resources are structured, they now allow the use of lifecycle rules, keeping costs down for
+running CDK applications over a long period of time.
+
+### Why should I use this feature?
+
+You should use this feature if you:
+
+- Want to take advantage of lifecycle rules on asset staging resources;
+- Do not use ECR and don't want to see the SecurityHub warning that tells you you have an empty ECR repository;
+- Need to deploy to multiple regions in a set of accounts and want to use StackSets to bootstrap the accounts;
```

Comment:
Why can't I do this today?

Reply:
You can but all Organizations-powered stacksets deploy in one region. So if you want:

- Acct 1, region A
- Acct 2, region B

You would need to add automatic stacksetting for region A and region B, leading you to bootstrap the full Cartesian product of `(1, A), (1, B), (2, A), (2, B)`.